src/plugins/intel_cpu/src/nodes/lora.cpp

```diff
@@ -86,24 +88,45 @@ void LoRA::selectOptimalPrimitiveDescriptor() {
     selectPrimitiveDescriptorByIndex(0);
 }
 
-// @todo add ascii diagram for memory mapping / reuse
-void LoRA::createPrimitive() {
+int LoRA::registerToAllocationContext(int offset, AllocationContext& context) {
     CPU_NODE_ASSERT(getOriginalInputsNumber() == m_graph.inputsNumber(),
                     "Number of node inputs must be equal the number of inner graph's inputs");
 
-    std::vector<MemoryPtr> inputMemory;
     for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
-        auto srcEdgeMem = getSrcMemoryAtPort(i);
-        auto mem = std::make_shared<Memory>(getEngine(), srcEdgeMem->getDescPtr(), srcEdgeMem->getMemoryBlock());
-        subgraphMemoryPtrs.push_back(mem);
-        inputMemory.emplace_back(std::move(mem));
+        auto parentEdge = getParentEdgeAt(i);
+        auto inputEdges = m_graph.getInputNodeByIndex(i)->getChildEdgesAtPort(0);
+        for (const auto& inputEdge : inputEdges) {
+            OPENVINO_ASSERT(inputEdge->getStatus() == Edge::Status::Uninitialized,
+                            "Expected Uninitialized Edge instead of: ",
+                            static_cast<int>(inputEdge->getStatus()));
+            inputEdge->sharedMemFrom(parentEdge);
+        }
     }
 
     CPU_NODE_ASSERT(getOriginalOutputsNumber() == m_graph.outputsNumber(),
                     "Number of node outputs must be equal the number of inner graph's outputs");
 
-    std::vector<MemoryPtr> outputMemory{getDstMemoryAtPort(0)};
-    m_graph.Activate(inputMemory, outputMemory);
+    for (size_t i = 0; i < getOriginalOutputsNumber(); i++) {
+        auto childEdge = getChildEdgeAt(i);
+        auto outputEdge = m_graph.getOutputNodeByIndex(i)->getParentEdgeAt(0);
+        outputEdge->sharedMemFrom(childEdge);
+    }
+
+    return m_graph.RegisterToAllocationContext(offset, context);
+}
+
+void LoRA::createPrimitive() {
+    CPU_NODE_ASSERT(getOriginalInputsNumber() == m_graph.inputsNumber(),
+                    "Number of node inputs must be equal the number of inner graph's inputs");
+    // Workaround to avoid making LoRa node always executable (isExecutable() = true)
+    // This way we update subgraph's input memory without performing an actual Infer() call
+    for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
+        auto subgraphInputNode = m_graph.getInputNodeByIndex(i);
+        auto subgraphInputMemory = subgraphInputNode->getDstMemoryAtPort(0);
+        subgraphMemoryPtrs.emplace_back(subgraphInputMemory);
```

Comment:
Are the memory objects valid before we call `Activate()`?

Reply:
same answer here

Reply:
same answer here